src/StoreApi/Routes/V1/CartUpdateCustomer.php

```diff
@@ -182,6 +182,19 @@ protected function get_route_post_response( \WP_REST_Request $request ) {
 				'shipping_phone'      => $shipping['phone'] ?? null,
 			)
 		);
+		// We want to only get additional fields passed, since core ones are already saved.
+		$core_fields = array( 'first_name', 'last_name', 'company', 'address_1', 'address_2', 'city', 'state', 'postcode', 'country', 'phone', 'email' );
+
+		$additional_shipping_values = array_diff_key( $shipping, array_flip( $core_fields ) );
+		$additional_billing_values  = array_diff_key( $billing, array_flip( $core_fields ) );
+
+		// We save them one by one, and we add the group prefix.
+		foreach ( $additional_shipping_values as $key => $value ) {
+			$this->additional_fields_controller->persist_field_for_customer( "/shipping/{$key}", $value, $customer );
+		}
+		foreach ( $additional_billing_values as $key => $value ) {
+			$this->additional_fields_controller->persist_field_for_customer( "/billing/{$key}", $value, $customer );
+		}
```

Comment:
Because additional billing and shipping fields will live alongside other fields, we filter core ones, leave only the additional ones, and save them.